infra/external/weather/weather.go

```diff
@@ -46,15 +42,32 @@ func NewWeather(conf repository.Config) (*Weather, error) {
 	return &Weather{
 		endpoint: conf.WeatherAPI(),
 		audience: audience,
-		client:   client,
 	}, nil
 }
 
+func (w *Weather) newTransport(ctx context.Context) (http.RoundTripper, error) {
+	ts, err := idtoken.NewTokenSource(ctx, w.audience)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to create token source: %w", err)
+	}
+
+	t, err := htransport.NewTransport(ctx, otelhttp.NewTransport(http.DefaultTransport), option.WithTokenSource(ts))
+	if err != nil {
+		return nil, xerrors.Errorf("failed to create idtoken client: %w", err)
+	}
+
+	return t, nil
+}
+
 // Fetch an weather map image.
 func (w *Weather) Fetch(ctx context.Context) (io.ReadCloser, error) {
-	cli, err := idtoken.NewClient(ctx, w.audience, idtoken.WithHTTPClient(w.client))
```

Comment:
`idtoken.NewClient` did not support `idtoken.WithHTTPClient`.
- https://github.com/googleapis/google-api-go-client/blob/v0.74.0/idtoken/idtoken.go#L56-L57
- https://github.com/googleapis/google-api-go-client/blob/v0.74.0/transport/http/dial.go#L58-L60

Reply:
## Document
https://cloud.google.com/run/docs/authenticating/service-to-service#acquire-token